### PR TITLE
Bai 1422 fix all pylance warnings in python client

### PR DESCRIPTION
**Pull request by IR96334**

### CODE REVIEW

File: lib/python/docs/notebooks/experiment_tracking_demo.ipynb [hunk 1]
```diff
@@ -211,7 +202,7 @@
    "source": [
     "# Setting local tracking URI and experiment name\n",
     "mlflow.set_tracking_uri(uri=\"http://127.0.0.1:5000\")\n",
-    "mlflow.set_experiment(\"Demonstator\")\n",
+    "mlflow.set_experiment(\"hgfc\")\n",
```
Comment (GB27247) [inline lib/python/docs/notebooks/experiment_tracking_demo.ipynb:205]: Maybe change this back to something less abstract?

File: lib/python/docs/notebooks/experiment_tracking_demo.ipynb [hunk 2]
```diff
@@ -238,7 +229,9 @@
    "outputs": [],
    "source": [
     "experiment_mlflow = model.create_experiment()\n",
-    "experiment_mlflow.from_mlflow(tracking_uri=\"http://127.0.0.1:5000\", experiment_id=\"\") # <- INSERT MLFLOW EXPERIMENT ID. CAN BE FOUND ON THE UI."
+    "experiment_mlflow.from_mlflow(\n",
+    "    tracking_uri=\"http://127.0.0.1:5000\", experiment_id=\"523964781059248151\"\n",
```
Comment (GB27247) [inline lib/python/docs/notebooks/experiment_tracking_demo.ipynb:233]: Could we redact this id and add a screenshot to show the user where this id coming from?

File: lib/python/docs/notebooks/experiment_tracking_demo.ipynb [hunk 3]
```diff
@@ -270,7 +263,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "experiment_mlflow.publish(mc_loc=\"performance.performanceMetrics\", run_id=\"\")"
+    "experiment_mlflow.publish(mc_loc=\"performance.performanceMetrics\", run_id=\"bcee7ec632894ca4a4208f105e7d9755\", semver=\"0.2.0\")"
```
Comment (GB27247) [inline lib/python/docs/notebooks/experiment_tracking_demo.ipynb:266]: Same with this could we redact this id and and use a screenshot?

**Review by GB27247 — CHANGES_REQUESTED** [review]

File: lib/python/src/bailo/helper/model.py [hunk 4]
```diff
@@ -614,5 +617,5 @@ def __select_run(self, select_by: str):
         ordered_runs = sorted(runs, key=lambda run: run["target"])
         if order_str == "MIN":
             return ordered_runs[0]
-        if order_str == "MAX":
+        else:  # MAX
```
[the remark this answers is not in the document]
Comment (IR96334) [reply lib/python/src/bailo/helper/model.py:620]: I did this to satisfy the returned dict type as with 2 if statements there could be a case where neither are satisfied thus a dict isn't returned, but there is a check before that ensures that order_str is either "MAX" or "MIN" so this isn't actually possible. I can think of 4 solutions:
1. Return dummy empty dict
2. Keep as else
3. Add ignore for pylance
4. return dict | None
Leaning more towards 1. and 3. but am honestly not sure what the best approach would be. Thoughts?

File: lib/python/docs/notebooks/experiment_tracking_demo.ipynb (hunk 1 above)
Comment (IR96334) [reply lib/python/docs/notebooks/experiment_tracking_demo.ipynb:205]: Done

File: lib/python/src/bailo/core/client.py [hunk 5]
```diff
@@ -454,12 +453,12 @@ def get_reviews(
         :param version: Model version, defaults to None
         :return: JSON response object.
         """
-        active = str(active).lower()
+        _active = repr(active).lower()
```
[the remark this answers is not in the document]
Comment (IR96334) [reply lib/python/src/bailo/core/client.py:456]: No idea. Changed

File: lib/python/docs/notebooks/experiment_tracking_demo.ipynb (hunk 3 above)
Comment (IR96334) [reply lib/python/docs/notebooks/experiment_tracking_demo.ipynb:266]: Done

File: lib/python/docs/notebooks/experiment_tracking_demo.ipynb (hunk 2 above)
Comment (IR96334) [reply lib/python/docs/notebooks/experiment_tracking_demo.ipynb]: Done

File: lib/python/docs/notebooks/datacards_demo.ipynb
[the remark this answers is not in the document]
Comment (IR96334) [reply lib/python/docs/notebooks/datacards_demo.ipynb:1]: Done.

File: lib/python/.gitignore [hunk 6]
```diff
@@ -132,6 +132,5 @@ dmypy.json
 # VS Code
 .vscode/
 
-# MLFlow
-mlartifacts/
-mlruns/
\ No newline at end of file
+#resnet
+/docs/notebooks/resnet50_weights.pth
```
Comment (GB27247) [inline lib/python/.gitignore:136]: Should this be an absolute path?

**Review by GB27247 — CHANGES_REQUESTED** [review]

File: lib/python/src/bailo/helper/model.py [hunk 7]
```diff
@@ -188,12 +188,12 @@ def from_mlflow(
...
 
         name = sel_model.name
-        description = sel_model.description + " Imported from MLFlow."
+        description = str(sel_model.description) + " Imported from MLFlow."
         bailo_res = client.post_model(
             name=name, kind=EntryKind.MODEL, description=description, team_id=team_id, visibility=visibility
         )
         model_id = bailo_res["model"]["id"]
-        logger.info(f"MLFlow model successfully imported to Bailo with ID %s.", model_id)
+        logger.info("MLFlow model successfully imported to Bailo with ID {model_id}.")
```
[the remark this answers is not in the document]
Comment (IR96334) [reply lib/python/src/bailo/helper/model.py:196]: Thanks, changed.

File: lib/python/src/bailo/helper/model.py [hunk 8]
```diff
@@ -535,6 +537,7 @@ def publish(
                 "Either select_by (e.g. 'accuracy MIN|MAX') or run_id is required to publish an experiment run."
             )
 
+        sel_run: dict[str, Any]
```
[the remark this answers is not in the document]
Comment (IR96334) [reply lib/python/src/bailo/helper/model.py:540]: Sure.

File: lib/python/src/bailo/helper/release.py [hunk 9]
```diff
@@ -265,13 +272,14 @@ def upload(self, path: str, data: BytesIO | None = None) -> str:
         with tqdm(
             total=size, unit="B", unit_scale=True, unit_divisor=BLOCK_SIZE, postfix=f"uploading {name}", colour=colour
         ) as t:
-            wrapped_buffer = CallbackIOWrapper(t.update, data, "read")
-            res = self.client.simple_upload(self.model_id, name, wrapped_buffer).json()
+            wrapped_buffer: BytesIO = CallbackIOWrapper(t.update, data, "read")  # type: ignore
```
[the remark this answers is not in the document]
Comment (IR96334) [reply lib/python/src/bailo/helper/release.py:275]: Changed

File: lib/python/src/bailo/helper/release.py [hunk 10]
```diff
@@ -238,23 +237,31 @@ def upload(self, path: str, data: BytesIO | None = None) -> str:
...
 
-            if is_zip:
+            if zip_required:
                 os.remove(path)
 
+        # cache the current file position then move to the end and get the size before moving it back.
         old_file_position = data.tell()
         data.seek(0, os.SEEK_END)
-        size = data.tell()
+        size: int = data.tell()
```
[the remark this answers is not in the document]
Comment (IR96334) [reply lib/python/src/bailo/helper/release.py:264]: Changed

File: lib/python/.gitignore (hunk 6 above)
Comment (IR96334) [reply lib/python/.gitignore:136]: Done

File: lib/python/src/bailo/helper/model.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (GB27247) [reply lib/python/src/bailo/helper/model.py]: Could we do this with all other instances of this :)

[the remark this answers is not in the document]
Comment (IR96334) [reply lib/python/src/bailo/helper/model.py]: Done

File: lib/python/src/bailo/helper/entry.py [hunk 11]
```diff
@@ -60,7 +69,7 @@ def card_from_schema(self, schema_id: str | None = None) -> None:
         res = self.client.model_card_from_schema(model_id=self.id, schema_id=schema_id)
         self.__unpack_card(res["card"])
 
-        logger.info(f"Card for ID %s successfully created using schema ID %s.", self.id, schema_id)
+        logger.info("Card for ID {self.id} successfully created using schema ID {schema_id}.")
```
Comment (GB27247) [inline lib/python/src/bailo/helper/entry.py:72]: Could we change this log back to an fstring

**Review by GB27247 — CHANGES_REQUESTED** [review]

File: lib/python/src/bailo/helper/entry.py (hunk 11 above)
Comment (IR96334) [reply lib/python/src/bailo/helper/entry.py:72]: Done

**Review by ARADDCC002 — APPROVED** [review]

**Pull request merged**